scripts/dev.sh

```
@@ -105,11 +105,26 @@ if [ ! -d "$VENV_PATH" ] && [ "$WT_ROOT" != "$MAIN_WT" ]; then
+  #
+  # Only ever sync the venv this worktree owns: syncing the main worktree's venv
+  # from a child would repoint its editable install at us (see the warning in
+  # scripts/lib/worktree.sh).  Non-fatal, because a lock that is stale mid
+  # dependency-edit should be visible but should not block the dev loop.
+  if command -v uv >/dev/null 2>&1 && [ "$VENV_PATH" = "${WT_ROOT}/.venv" ]; then
+    if ! (cd "$WT_ROOT" && UV_PROJECT_ENVIRONMENT="$VENV_PATH" uv sync --locked --quiet); then
+      echo "WARNING: 'uv sync --locked' failed — .venv may not match uv.lock." >&2
+      echo "         If you have edited pyproject.toml, run 'uv lock'." >&2
+    fi
+  fi
   source "$VENV_PATH/bin/activate"
```

Shellcheck (0.11.0)

[info] Not following: ./bin/activate was not specified as input (see shellcheck -x).

(SC1091)